pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/scheduler/NamespaceUnloadScheduler.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.pulsar.broker.loadbalance.extensions.channel.ServiceUnitStateChannel;
+import org.apache.pulsar.broker.loadbalance.extensions.models.UnloadDecision;
+import org.apache.pulsar.common.util.FutureUtil;
+import org.apache.pulsar.common.util.Reflections;
+
+@Slf4j
+public class NamespaceUnloadScheduler implements LoadManagerScheduler {
+
+    private final NamespaceUnloadStrategy namespaceUnloadStrategy;
+
+    private final ScheduledExecutorService loadManagerExecutor;
+
+    private final LoadManagerContext context;
+
+    private final ServiceUnitStateChannel channel;
+
+    private final ServiceConfiguration conf;
+
+    private volatile ScheduledFuture<?> loadSheddingTask;
+
+    private final Map<String, Long> recentlyUnloadedBundles;
+
+    private final Map<String, Long> recentlyUnloadedBrokers;
+
+    private volatile CompletableFuture<Void> currentRunningFuture = null;
+
+    public NamespaceUnloadScheduler(ScheduledExecutorService loadManagerExecutor,
+                                    LoadManagerContext context,
+                                    ServiceUnitStateChannel channel) {
+        this(loadManagerExecutor, context, channel, createNamespaceUnloadStrategy(context.brokerConfiguration()));
+    }
+
+    @VisibleForTesting
+    protected NamespaceUnloadScheduler(ScheduledExecutorService loadManagerExecutor,
+                                       LoadManagerContext context,
+                                       ServiceUnitStateChannel channel,
+                                       NamespaceUnloadStrategy strategy) {
+        this.namespaceUnloadStrategy = strategy;
+        this.recentlyUnloadedBundles = new ConcurrentHashMap<>();
+        this.recentlyUnloadedBrokers = new ConcurrentHashMap<>();
+        this.loadManagerExecutor = loadManagerExecutor;
+        this.context = context;
+        this.conf = context.brokerConfiguration();
+        this.channel = channel;
+    }
+
+    @Override
+    public synchronized void execute() {
+        boolean debugMode = conf.isLoadBalancerDebugModeEnabled() || log.isDebugEnabled();
+        if (debugMode) {
+            log.info("Load balancer enabled: {}, Shedding enabled: {}.",
+                    conf.isLoadBalancerEnabled(), conf.isLoadBalancerSheddingEnabled());
+        }
+        if (currentRunningFuture != null && !currentRunningFuture.isDone()
+                || !(conf.isLoadBalancerEnabled() && conf.isLoadBalancerSheddingEnabled())) {
+            return;
+        }
+        this.currentRunningFuture = channel.isChannelOwnerAsync().thenCompose(isChannelOwner -> {
+            if (!isChannelOwner) {
+                return CompletableFuture.completedFuture(null);
+            }
+            return context.brokerRegistry().getAvailableBrokersAsync().thenCompose(availableBrokers -> {
+                if (debugMode) {
+                   log.info("Available brokers: {}", availableBrokers);
+                }
+                if (availableBrokers.size() <= 1) {
```

Comment:
Can we move this logic to TransferShedder?( and accordingly emit the decision result)

Reply:
Why do we need to move to `TransferShedder`? Are there any benefits? If only one broker is available, we don't need to do to unload.

Reply:
Ok this makes sense.